mantidproject/mantid
# Project Recovery: Handle Legacy checkpoints

~~ONLY TO BE MERGED AFTER #23737 AND #23638~~

**Description of work.**
I rearranged some of project recovery's code to reduce calculations and added handling for legacy checkpoints, which should stop people getting a pop up at mantid startup asking if you want to terminate.

**To test:**
[computername.zip](https://github.com/mantidproject/mantid/files/2459958/computername.zip)
- Extract this zip into your ~/.mantid/recovery/YOURCOMPUTERNAME directory. (Make sure these files are the only ones there)
- Start mantid
- All these files should dissapear and nothing should pop up for mantid at start for project recovery.

The folder 123 represents a folder with a lock file in it,
The folder 2018-11-01T10-10-11 represents a legacy checkpoint,
The folder 121345 represents an empty folder with no checkpoints inside it.



Fixes #23739. 


This doesn't need release notes as it fixes a bug added after the last release

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
